### DISCUSSION TITLE
Lookup user ID from external Authorized provider ID

### DISCUSSION THREAD

**Discussion by kisamoto** [pocketbase/pocketbase#5651] [Q&A]
Is it possible to do a reverse lookup from an Authorized provider ID to get a user ID in the web UI?

For example I have a user logged in with an OAuth provider where the provider ID is `9c491513-cd03-4a20-b1b3-bca6c93cc7c1`. Can I somehow write a query in the users table to find the user record that matches that ID?

Thanks & happy hacking

**Comment by ganigeorgiev** [comment]
Not directly. In PocketBase < v0.23 the links are stored in a `_externalAuths` table but it is not backed by a collection (_this is no longer the case with PocketBase v0.23+_).

For now, one possible option could be to create a helper admin-only View collection with query like:

```sql
SELECT id, recordId, provider, providerId FROM "_externalAuths"
```

And then from the users Admin UI search bar you should be able to search with expression like:

```js
@collection.linkedUsersView.recordId = id && @collection.linkedUsersView.providerId = "9c491513-cd03-4a20-b1b3-bca6c93cc7c1"
```
